cloudmetrics/metrics/fourier.py

```diff
@@ -0,0 +1,573 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import fftpack, ndimage, linalg
+from scipy.optimize import curve_fit
+from ..utils import compute_r_squared
+
+
+def _get_rad(data):
+    # From https://medium.com/tangibit-studios/2d-spectrum-characterization-e288f255cc59
+    h = data.shape[0]
+    hc = h // 2
+    w = data.shape[1]
+    wc = w // 2
+
+    # create an array of integer radial distances from the center
+    Y, X = np.ogrid[0:h, 0:w]
+    r = np.hypot(X - wc, Y - hc)
+
+    return r
+
+
+def _get_psd_1d_radial(psd_2d, dx):
+
+    # TODO: Assumes even number of points in psd_2d and square domain
```

Comment:
Could you put a shape for the check here and raise `NotImplementedError` if there's an issue?

```python
nx, ny = psd_2d.shape
if nx != ny:
    raise NotImplementedError(f"nx != ny ({nx} != {ny})")`?
```

Reply:
Good idea, I've stuck it in ``compute_spectra`` itself, as this assumption is true for azimuthal spectra as well. By the way, supporting metrics on non-square domains is functionality that I'd like to implement sometime after we're done with the refactoring bit :)